lib/Conversion/TorchToTosa/TorchToTosa.cpp

```diff
@@ -1132,6 +1133,34 @@ Type getMatMulOutputType(Type inputElemTy, PatternRewriter &rewriter) {
   return outputElemTy;
 }
 
+RankedTensorType getCastedInputTypeForMatmul(Value inputValue,
+                                             PatternRewriter &rewriter) {
+  // Check to see if the inputs to the matmul where casted from another type
+  auto preCastType =
+      TypeSwitch<Operation *, RankedTensorType>(inputValue.getDefiningOp())
+          .Case([](AtenToDtypeOp op) {
+            return cast<RankedTensorType>(op->getOperand(0).getType());
+          })
+          .Case([](tosa::CastOp op) {
+            return cast<RankedTensorType>(op->getOperand(0).getType());
+          })
+          .Default([](Operation * /*op*/) { return RankedTensorType(); });
+  if (!preCastType) {
+    return preCastType;
+  }
+  // Calculate the expected accumulator type based on the input type of the cast
```

Comment:
Well for that case the pattern will not match because at the return the output type of the cast is checked against the calculated accumulator type. So for:
```
cast f32 -> f16
```
We calculate an accumulator type of `f32` which does not equal what the cast does (i.e. `f16`). So the helper function just returns an empty type. Meaning we do not add the casts or change the element types. 

I will add a test case to show this.

Reply:
done